src/mapping.ts

```diff
@@ -0,0 +1,275 @@
+  mergeTimesPersist.push(event.block.timestamp.toI32());
+  nftPersist.mergeTimes = mergeTimesPersist;
+
+  // mergeCount, massSize, color, tier of nftPersist
+  nftPersist.mergeCount++;
+  nftPersist.massSize = (updatedValue % CLASS_MULTIPLIER) as BigInt;
+  const bigIntTier: BigInt = updatedValue / CLASS_MULTIPLIER;
+  const tier: string = checkMergeClass(bigIntTier);
+  nftPersist.tier = tier;
+  const alphaCheck = nftPersist.isAlpha;
+  nftPersist.color = checkColor(tier, alphaCheck);
+
+  nftBurned.save();
+  nftPersist.save();
+
+  // TODO: when decodeValue() works, may have to tweak as I haven't looked at it in a bit.
+
+  // nftPersist.massSize = decodeValue(
+  //   updatedValue
+  // ).massSize;
+  // nftPersist.color = decodeValue(updatedValue).color;
+  // nftPersist.tier = decodeValue(updatedValue).tier;
+
+  // TODO: FYI: only other time MassUpdate happens is when _burnNoEmitTransfer happens, I realize now that I need to make sure this accounts for that, but I do not think I have seen the function called tbh in etherscan / don't understand when it would get called.
+}
+
+/* ========== CALL HANDLERS ========== */
+
+export function handleWhitelistUpdate(call: WhitelistUpdateCall): void {
+  const address = call.inputs.address_.toHexString();
+  const status = call.inputs.status;
+
+  const user = User.load(address) as User;
+
+  user.whitelist = status;
+  user.save();
+}
+
+/* ========== GENERAL FUNCTIONS ========== */
+
+/**
+ * @notice load NFT or call createNFT() for new NFT
+ * @param event: Transfer from handleTransfer() caller
+ * @returns NFT with updated owner, or newly minted NFT entity
+ */
+function updateNFT(tokenId: string, userId: string): NFT {
+  const nft = NFT.load(tokenId) as NFT;
+  nft.owner = userId;
+  return nft;
+}
+
+/**
+ * @notice create new NFT entity types upon mint
+ * @param event: Transfer from handleTransfer() caller function
+ * @dev key thing is the nftValue variable that is the encoded value metric used within the smart contracts to encode the mass and class data for each nft
+ */
+export function createNFT(event: Transfer, userId: string): NFT {
+  const tokenIdString = event.params.tokenId.toString();
+  const tokenId = event.params.tokenId;
+  const nft = new NFT(tokenIdString);
+  const contract = Merge.bind(event.address);
+  nft.owner = userId;
+
+  const nftValue = contract.getValueOf(tokenId);
+
+  // obtain nft fields throughs calculating off of encoded value
+  if (nftValue >= NOT_WHITE_VALUE) {
+    log.info("nft value: {}", [nftValue.toString()]);
+  }
+  nft.massValue = nftValue;
+  nft.massSize = (nftValue % CLASS_MULTIPLIER) as BigInt;
+  const bigIntTier: BigInt = nftValue / CLASS_MULTIPLIER;
+  const tier: string = checkMergeClass(bigIntTier);
+  nft.tier = tier;
+  nft.isAlpha = false;
+  const alphaCheck = nft.isAlpha;
+  nft.color = checkColor(tier, alphaCheck);
+
+  nft.mergeCount = 0;
+
+  nft.absorbedNFTs.push("n/a");
+  nft.mergeTimes.push(0);
```

Comment:
this is not proper, i am surpirsed this compiled.

First try to delete these, and see of it compiles. i think it should but i am like 80% sure.  if it does not compile, do this:

```
  nft.absorbedNFTs = [];
  nft.mergeTimes = [];
```

Reply:
It compiles even when I comment it out. So I will delete it and we should be good!